cmd/datamon/cmd/config_create.go

```diff
@@ -0,0 +1,78 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+)
+
+var configGen = &cobra.Command{
+	Use:   "create",
+	Short: "Create a local config file",
+	Long: `Creates a local config file to use for datamon to hold flags that do not change, like remote config bucket or current context to use.
+
+	By default, this configuration file will be placed in ` + configFileLocation(false) + `.
+
+	Use the DATAMON_CONFIG environment variable to change this default target.
+	`,
+	Example: `# Replace path to gcloud credential file. Use absolute path
+% datamon config create --credential /Users/ritesh/.config/gcloud/application_default_credentials.json,
+config file created in /Users/ritesh/.datamon2/datamon.yaml
+
+# Replace path to gcloud credentials file (use absolute path here)
+% datamon config create --credential /Users/ritesh/.config/gcloud/application_default_credentials.json
+config file created in /Users/ritesh/.datamon2/datamon.yaml
+
+# Specify a config bucket to store context details
+% datamon config create --config fred-datamon-config --context test-context
+config file created in /Users/ritesh/.datamon2/datamon.yaml
+
+# Generate config in some non-default location
+% ` + envConfigLocation + `=~/.config/.datamon/config.yaml datamon config create --config "remote-config-bucket"
+config file created in /Users/ritesh/.config/.datamon/config.yaml
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		_, err := paramsToContributor(datamonFlags)
+		if err != nil {
+			wrapFatalln("contributor datamonFlags present", err)
+			return
+		}
+
+		config := CLIConfig{
+			Config:     datamonFlags.core.Config,
+			Context:    datamonFlags.context.Descriptor.Name,
+			Credential: datamonFlags.root.credFile,
+		}
+		o, err := config.MarshalConfig()
```

Comment:
@ransomw1c hopes this now addresses your concerns about whether to yaml or not to yaml

Reply:
have added a comment in `MarshallConfig` to continue this nitpick

Reply:
wanted to be consistent with the MarshalXXX way to produce YAML docs adopted in pkg/models